banks-server/src/banks_server.rs

```diff
@@ -331,6 +333,58 @@ impl Banks for BanksServer {
             .await
     }
 
+    async fn process_versioned_transaction_with_commitment_and_context(
+        self,
+        _: Context,
+        transaction: VersionedTransaction,
+        commitment: CommitmentLevel,
+    ) -> BanksTransactionResultWithMetadata {
+        let bank = self.bank(commitment);
+        let batch = match bank.prepare_entry_batch(vec![transaction]) {
+            Ok(batch) => batch,
+            Err(err) => {
+                return BanksTransactionResultWithMetadata {
+                    result: Err(err),
+                    metadata: None,
+                }
+            }
+        };
+
+        let (
+            TransactionResults {
+                mut execution_results,
+                ..
+            },
+            ..,
+        ) = bank.load_execute_and_commit_transactions(
```

Comment:
nit: up to you, but rather than calling `load_execute_and_commit_transactions` directly, you could add a new function on the bank, similar to `process_transaction_with_logs`, which works with just one transaction and then gives just those execution results

Reply:
Good call. I changed `process_transaction_with_logs` to `process_transaction_with_metadata`